### PR TITLE
[ioctl] Refactor version command in new ioctl

### LINKED ISSUE #3278: [ioctl] refactor version command in new ioctl
### What would you like to be added
 - Step 1: refactor  command  [newcmd/version/version.go](https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/version/version.go) 
 - Step 2: append necessary unit test case in [newcmd/version/version_test.go](https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/version/version_test.go) 

#### Notes:
1. Use [function of client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) . 
2. [Output package](https://github.com/iotexproject/iotex-core/blob/master/ioctl/output) is deprecated. `e.g.` `output.PrintError` could be replaced with `errors.Wrap`.
3. Global variables (config.ReadConfig  config.DefaultConfigFile etc.) have been wrapped into client interface. Please use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to access them if needed
4. The logic of new code should be similar to [the old one](https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/version/version.go ).
5. In each unit test, the results should be checked whether they are as the expected. 

#### Example:
1. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountinfo.go
2. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountdelete.go
3. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountcreateadd.go

### Why is this needed
Unit tests couldn't be added for the old ioctl command (./ioctl/cmd)

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description

Refactor `version` command in new `ioctl`, with the following note.

* Output package is deprecated, replace it with errors package.
* Replace fmt.Println with cmd.Println
* Refactor unit test to cover the modification.

Fixes #3278 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] TestVersionCommand

**Test Configuration**:

- Firmware version: Ubuntu 21.04
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ioctl/newcmd/version/version.go [hunk 1]
```diff
@@ -51,8 +51,41 @@ func NewVersionCmd(cli ioctl.Client) *cobra.Command {
...
+				Object: "Client",
+				VersionInfo: &iotextypes.ServerMeta{
+					PackageVersion:  ver.PackageVersion,
+					PackageCommitID: ver.PackageCommitID,
+					GitStatus:       ver.GitStatus,
+					GoVersion:       ver.GoVersion,
+					BuildTime:       ver.BuildTime,
+				},
+			}
+			cmd.Println(fmt.Sprintf("%s:\n%+v\n", message.Object, message.VersionInfo))
```
Comment (CoderZhi) [inline ioctl/newcmd/version/version.go:64]: `cmd.Printf("%s:\n%+v\n", "Client", &iotextypes.ServerMeta{...})`

File: ioctl/newcmd/version/version.go [hunk 2]
```diff
@@ -51,8 +51,41 @@ func NewVersionCmd(cli ioctl.Client) *cobra.Command {
...
+				if sta, ok := status.FromError(err); ok {
+					return errors.New(sta.Message())
+				}
+				return errors.Wrap(err, "failed to get version from server")
+			}
+			message = versionMessage{
+				Object:      config.ReadConfig.Endpoint,
+				VersionInfo: response.ServerMeta,
+			}
+			cmd.Println(fmt.Sprintf("%s:\n%+v\n", message.Object, message.VersionInfo))
```
Comment (CoderZhi) [inline ioctl/newcmd/version/version.go:87]: same here

File: ioctl/newcmd/version/version.go [hunk 3]
```diff
@@ -51,8 +51,41 @@ func NewVersionCmd(cli ioctl.Client) *cobra.Command {
...
+			apiClient, err := cli.APIServiceClient(ioctl.APIServiceConfig{
+				Endpoint: endpoint,
+				Insecure: insecure,
+			})
+			if err != nil {
+				return err
+			}
+
+			response, err := apiClient.GetServerMeta(
```
Comment (huof6829) [inline ioctl/newcmd/version/version.go:73]: add 
https://github.com/iotexproject/iotex-core/blob/651d77bb39b759614627003f9a2441fe0ff3bb8b/ioctl/cmd/version/version.go#L92-L95

**Review by huof6829 — APPROVED** [review]

File: ioctl/newcmd/version/version.go [hunk 4]
```diff
@@ -34,84 +36,63 @@ var (
...
+			}
+			cmd.Println(fmt.Sprintf("%s:\n%+v\n", c.Config().Endpoint, response.ServerMeta))
+			return nil
 		},
 	}
 	vc.PersistentFlags().StringVar(
 		&endpoint,
 		"endpoint",
-		cli.Config().Endpoint,
+		c.Config().Endpoint,
```
Comment (Liuhaai) [inline ioctl/newcmd/version/version.go:88]: `c` -> `cli`

**Review by Liuhaai — APPROVED** [review]

**Pull request merged**